packages/xo-web/src/common/xo/index.js

```diff
@@ -821,13 +821,38 @@ export const setRemoteSyslogHost = (host, syslogDestination) =>
 export const setRemoteSyslogHosts = (hosts, syslogDestination) =>
   Promise.all(map(hosts, host => setRemoteSyslogHost(host, syslogDestination)))
 
-export const restartHost = (host, force = false, suspendResidentVms = false) =>
+export const restartHost = (host, force = false, suspendResidentVms = false, forceSmartReboot = false) =>
   confirm({
     title: _('restartHostModalTitle'),
     body: _('restartHostModalMessage'),
   }).then(
     () =>
-      _call('host.restart', { id: resolveId(host), force, suspendResidentVms })
+      _call('host.restart', { id: resolveId(host), force, suspendResidentVms, forceSmartReboot })
+        .catch(async error => {
+          if (
+            suspendResidentVms &&
+            !forceSmartReboot &&
+            operationBlocked.is(error, {
+              code: 'suspend',
+            })
+          ) {
+            await confirm({
+              body: (
+                <p>
+                  <Icon icon='alarm' /> {_('forceSmartRebootHost', { nVms: error.data.objectId.length })}
+                </p>
+              ),
+              title: _('restartHostModalTitle'),
+            })
+            forceSmartReboot = true
+            return _call('host.restart', {
+              id: resolveId(host),
+              force,
+              suspendResidentVms,
+              forceSmartReboot,
+            })
+          }
+        })
         .catch(async error => {
```

Comment:
`error` can be `undefined` if you cancel the confirm modal. I guess using `await/catch` instead of `.catch()` can help simplify the chain and fix this.

Reply:
I finally decided to keep the `.catch()` methods and test the `undefined` error case directly in them. This minimizes changes and avoids having nested `try/catch`.